### PR TITLE
[clang][NFC] Add documentation for `CastExpr::path()`.

### PR DESCRIPTION
**Pull request by martinboehme**
This didn't have any documentation, so I had to do some experimenting in godbolt when I used this in https://github.com/llvm/llvm-project/pull/84138, and my reviewer later also had some [questions](https://github.com/llvm/llvm-project/pull/84138#discussion_r1524855434) about this, so I figured it would be worth adding documentation.

### CODE REVIEW

**Review by ymand — APPROVED** [review]
thanks!

File: clang/include/clang/AST/Expr.h [hunk 1]
```diff
@@ -3552,6 +3552,15 @@ class CastExpr : public Expr {
   /// function that it invokes.
   NamedDecl *getConversionFunction() const;
 
+  /// Path through the class hierarchy taken by a `DerivedToBase` or
+  /// `UncheckedDerivedToBase` cast. For each derived-to-base edge in the path,
+  /// the path contains a `CXXBaseSpecifier` for the base class of that edge;
+  /// the entries are ordered from derived class to base class.
```
Comment (rjmccall) [inline clang/include/clang/AST/Expr.h:3558]: You can see in `CastConsistency` the set of cast kinds that require a base path; it's more than just these two.

Comment (martinboehme) [reply clang/include/clang/AST/Expr.h]: Thanks for pointing this out! I've fixed this -- WDYT?

**Comment by tbaederr** [comment]
Does this actually show up at the right place? Or is it now the documentation for `path_empty()`? Or  even `path_iterator`?

**Comment by martinboehme** [comment]
> Does this actually show up at the right place? Or is it now the documentation for `path_empty()`? Or even `path_iterator`?

Hm, good question. This seemed the most logical place to put the documentation in the source code, but of course, once it gets processed by Doxygen, it will probably get attached to something we don't want.

I've moved the documentation down, right above the declaration of `path()`. WDYT?

**Comment by tbaederr** [comment]
> I've moved the documentation down, right above the declaration of `path()`. WDYT?

I think that should work.

**Review by rjmccall — APPROVED** [review]

**Pull request merged**